python/lib/modeldata/dmod/modeldata/data/catchment_data.py

```diff
@@ -0,0 +1,88 @@
+import numpy as np
+
+from datetime import datetime
+from hypy import Catchment
+from typing import List, Optional
+
+
+class CatchmentData:
+    """
+    Encapsulation of forcing data for a particular catchment over some period.
```

Comment:
It should (or at least there's no reason it can't) be a mix of any required catchment data that fits into a tabular layout, where each row has the data points applicable to some period of time.  I'll adjust the doc string a little.